### PR TITLE
virtio_serial:Hot-unplug port and chardev and pci

### PR DESCRIPTION
**Pull request by nanliu-r**
ID: 1753608

Signed-off-by: Nana Liu <nanliu@redhat.com>

### CODE REVIEW

**Comment by nanliu-r** [comment]
Test results: --- the error of Linux is a known product bug, doesn't impact the auto result.


 (1/1) Host_RHEL.m8.u2.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.virtio_port_hotplug.hotplug_port_pci.without_console: PASS (94.91 s)

 (1/1) Host_RHEL.m8.u2.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.2.0.x86_64.io-github-autotest-qemu.virtio_port_hotplug.hotplug_port_pci.without_console: PASS (90.38 s)

 (1/1) Host_RHEL.m8.u2.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.2.0.x86_64.io-github-autotest-qemu.virtio_port_hotplug.hotplug_port_pci.without_console.q35: ERROR: QMP command 'device_del' failed    (arguments: {'id': 'vs1'},    error message: {'class': 'DeviceNotFound', 'desc': "Device 'vs1' not found"}) (99.55 s)

**Comment by nanliu-r** [comment]
@vivianQizhu @xiangchunfu Please help to review this, thanks a lot.

File: qemu/tests/virtio_serial_hot_unplug_port_pci.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+    :param test:   QEMU test object.
+    :param params: Dictionary with the test parameters.
+    :param env:    Dictionary with test environment.
+    """
+    vm = env.get_vm(params["main_vm"])
+    unplug_pci = params.get("unplug_pci")
+    os_type = params["os_type"]
+    sleep_time = int(params.get("sleep_time"))
+    repeat_times = int(params.get("repeat_times"))
```
Comment (vivianQizhu) [inline qemu/tests/virtio_serial_hot_unplug_port_pci.py]: Is there any other test cases use this script? Will `unplug_pci` and `repeat_times` have other values?

Comment (nanliu-r) [reply qemu/tests/virtio_serial_hot_unplug_port_pci.py]: Yes, when `unplug_pci = no` we won't hot-unplug pci. And the default value of `repeat_times` is 100.

Comment (nanliu-r) [reply qemu/tests/virtio_serial_hot_unplug_port_pci.py]: > Is there any other test cases use this script? Will `unplug_pci` and `repeat_times` have other values?

sorry, default value of  `repeat_times` is 1.

Comment (nanliu-r) [reply qemu/tests/virtio_serial_hot_unplug_port_pci.py]: > Is there any other test cases use this script? Will `unplug_pci` and `repeat_times` have other values?

Seems they are not the must so I delete them, thanks.

File: qemu/tests/cfg/virtio_port_hotplug.cfg [hunk 2]
```diff
@@ -37,6 +37,12 @@
             filesize = 100
             unplug_chardev_vc1 = no
             unplug_chardev_vc2 = no
+        - unplug_port_chardev_pci:
+            type = virtio_serial_unplug_port_chardev_pci
+            sleep_time = 0
+            start_vm = yes
```
Comment (yanan-fu) [inline qemu/tests/cfg/virtio_port_hotplug.cfg]: No need, it is the default value in framwork

File: qemu/tests/cfg/virtio_port_hotplug.cfg [hunk 3]
```diff
@@ -37,6 +37,12 @@
             filesize = 100
             unplug_chardev_vc1 = no
             unplug_chardev_vc2 = no
+        - unplug_port_chardev_pci:
+            type = virtio_serial_unplug_port_chardev_pci
+            sleep_time = 0
```
Comment (yanan-fu) [inline qemu/tests/cfg/virtio_port_hotplug.cfg]: For linux guest,  `0` is enough ?  I see some other chardev hot plug may wait for `0.5s` then plug the virtio port to this chardev.

File: qemu/tests/cfg/virtio_port_hotplug.cfg (hunk 2 above)
Comment (nanliu-r) [reply qemu/tests/cfg/virtio_port_hotplug.cfg]: thanks and update

File: qemu/tests/cfg/virtio_port_hotplug.cfg (hunk 3 above)
Comment (nanliu-r) [reply qemu/tests/cfg/virtio_port_hotplug.cfg]: ok, let's set the default value to 0.5s

**Comment by yanan-fu** [comment]
LGTM

please provide the latest test result, thanks!

**Comment by nanliu-r** [comment]
> LGTM
> 
> please provide the latest test result, thanks!

Thanks.

Test results:
 (10/10) repeat10.Host_RHEL.m7.u9.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.8.x86_64.io-github-autotest-qemu.virtio_port_hotplug.unplug_port_chardev_pci: PASS (79.63 s)
RESULTS    : PASS 10 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0

**Pull request merged**